python/tests/test_adjoint_solver.py

```diff
@@ -1060,6 +1060,25 @@ def test_periodic_design(self):
 
             print(f"PASSED: filter function = {selected_filter.__name__}")
 
+    def test_reverse_design(self):
+        """Verifies that the reverse_design on a given structure
+        finds initialization close to what it found previously."""
+        print("*** TESTING reverse_design ***")
+
+        data_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "data"))
+        expected = np.load(os.path.join(data_dir, "mpa_reverse_design.npy"))
+        target = np.load(os.path.join(data_dir, "mpa_reverse_design_target.npy"))
+
+        def processing(x):
+            filtered_field = mpa.conic_filter(x, 0.1, 2, 2, 200)
+            projected_field = mpa.tanh_projection(filtered_field, 8, 0.5)
+            return projected_field.flatten()
+
+        reverse_x = mpa.reverse_design(target, processing)
+        tol = 1e-5
```

Comment:
Is this value supposed to match the `ftol = 1e-5` in `reverse_design.py`?

Reply:
Not really. This value is just a small number to ensure future versions will have equally satisfying performance. I change it to a smaller value.